### ISSUE TITLE
Editor doesn't work with .NET Standard 2 nugets

### ISSUE THREAD

**Issue by praeclarum** [xamarin/xamarin-macios#4092]
### Steps to Reproduce

1. Create a Xamarin.iOS project
2. Add `System.Collections.Immutable` 1.4
3. Use `ImmutableArray.RemoveAll`
3. Note the editor errors
4. Manually reference .NET Standard
5. Editor will still have errors complaining about `System.Predicate` and `System.Object`

### Expected Behavior

I can use .NET Standard libraries.

### Actual Behavior

Missing reference errors all over the place

<img width="807" alt="screenshot 2018-05-15 13 08 53" src="https://user-images.githubusercontent.com/323548/40081759-d2c2a4b8-5843-11e8-93b8-1cc717115a54.png">

### Environment

```
=== Visual Studio Enterprise 2017 for Mac (Preview) ===

Version 7.6 Preview (7.6 build 711)
Installation UUID: 4ba352e1-73e8-4dab-a9db-28012ca05af8
Runtime:
	Mono 5.12.0.226 (2018-02/9824e260f56) (64-bit)
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 4.4.0.36 (master / 0c7c49a6)

	Package version: 512000226

=== NuGet ===

Version: 4.3.1.4445

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.0-preview2-26406-04
	2.0.3
SDK: /usr/local/share/dotnet/sdk/2.1.300-preview2-008530/Sdks
SDK Versions:
	2.1.300-preview2-008530
	2.0.3
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/msbuild/15.0/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.3
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Apple Developer Tools ===

Xcode 9.3.1 (14154.1)
Build 9E501

=== Xamarin.Mac ===

Version: 4.5.0.280 (Visual Studio Enterprise)
Hash: e85a7be8
Branch: master
Build date: 2018-05-03 11:35:01-0400

=== Xamarin.Android ===

Version: 8.3.0.19 (Visual Studio Enterprise)
Android SDK: /Users/fak/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		7.0 (API level 24)
		7.1 (API level 25)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 26.0.2
SDK Build Tools Version: 26.0.3

Java SDK: /usr
java version "1.8.0_152"
Java(TM) SE Runtime Environment (build 1.8.0_152-b16)
Java HotSpot(TM) 64-Bit Server VM (build 25.152-b16, mixed mode)

Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Xamarin.iOS ===

Version: 11.11.0.280 (Visual Studio Enterprise)
Hash: e85a7be8
Branch: master
Build date: 2018-05-03 11:35:00-0400

=== Xamarin Inspector ===

Version: 1.4.0
Hash: b3f92f9
Branch: master
Build date: Fri, 19 Jan 2018 22:00:34 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 706000711
Git revision: 634060c6c105fd3ee36678fe547680fd258cbf45
Build date: 2018-05-04 16:11:06-04
Xamarin extensions: c6720450018cb69a3da7c6d0aa0715c013783595
Build lane: monodevelop-lion-master

=== Operating System ===

Mac OS X 10.13.3
Darwin 17.4.0 Darwin Kernel Version 17.4.0
    Tue Dec 19 10:52:05 PST 2017
    root:xnu-4570.47.16~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

MSBuild Editor 2.0.3
Internet of Things (IoT) development (Preview) 7.5
```

### Example Project (If Possible)

[NoNetStd.zip](https://github.com/xamarin/xamarin-macios/files/2006612/NoNetStd.zip)

> VS bug [#617303](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/617303)

**Comment by mhutch** [comment]
Is this just in VS for Mac editor, and not at build time or on Windows?

VS/VSMac call MSBuild targets to get the list of referenced assemblies to use for intellisense, but I don't think the target VSMac currently uses includes the target that injects netstandard facade references if needed.

cc @mrward

**Comment by praeclarum** [comment]
This is just in the VS Mac editor (and error window). I can build and deploy.

Not sure what changed because I've used this library before in projects. Not sure if they changed the packaging or if VSM changed its resolution.

**Comment by mhutch** [comment]
Where you using the same version before? Older versions didn't contain netstandard 2.0 binaries.

**Comment by praeclarum** [comment]
My previous apps use v1.2 of Immutable.

**Comment by mhutch** [comment]
Yep looks like 1.2 contains netstandard 1.0 and PCL binaries, while 1.4 also contains netstandard2.0 binaries, which will take precedence for Xamarin.iOS.

**Comment by praeclarum** [comment]
@spouliot I'm curious why this didn't make the cut for 15-8 - supporting .NET Standard seems pretty important since all libraries are moving to it.

**Comment by mrward** [comment]
This is a VS Mac bug since it builds the NoNetStd project builds without any errors using msbuild or VS Mac.

When you compile with VS Mac or MSBuild the facade assemblies are referenced which includes netstandard.dll - https://gist.github.com/mrward/de7555b19c98326cfd4ae04c9255df2a

VS Mac uses the ResolveAssemblyReferences MSBuild target and that returns 6 assemblies:

```
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/mscorlib.dll
~/Projects/NoNetStd/packages/System.Collections.Immutable.1.4.0/lib/netstandard2.0/System.Collections.Immutable.dll
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.Core.dll
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.dll
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/System.Xml.dll
/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/mono/Xamarin.iOS/Xamarin.iOS.dll
```

So without the netstandard.dll returned VS Mac's text editor shows errors about a missing netstandard assembly reference.

Currently VS Mac [adds the facade assemblies](https://github.com/mono/monodevelop/blob/d14d691dba89cba6e5c198d3d9b960c7f47764b9/main/src/core/MonoDevelop.Core/MonoDevelop.Projects/DotNetProject.cs#L951-L974) if one the following is true:

1. The project references a .NET Standard project.
2. The project references a PCL project.
3. One of the assemblies returned from ResolveAssemblyReferences references System.Runtime.

Mikayla's [Unify project & assembly ref resolution pull request](https://github.com/mono/monodevelop/pull/2445) may be a fix here - it uses the ResolveReferences MSBuild target. An alternative quick fix here might be just to check for netstandard being referenced by any of the referenced assemblies as well as System.Runtime.

**Comment by mhutch** [comment]
Yes, my patch would fix this. But it's been sitting around for a year and has bitrotted, so a quick fix is probably a good idea.

I've been working piecemeal on resurrecting the patch. The parts that remove xbuild support and make the MSBuild engine mandatory are now merged in master, but there's a fair bit more to do.

**Comment by praeclarum** [comment]
@mrward I can confirm that referencing a .NET Std project makes the squiggles go away. Thank you for that workaround!

**Comment by mrward** [comment]
A fix for this has been included in VS Mac 7.6.0.1755 (release-7.6).

**Comment by praeclarum** [comment]
@mrward Thank you!
